packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -147,10 +147,50 @@ export const askEnvironmentVariableCarryOut = async (context: $TSContext, curren
 
   await envVarQuestion();
   Object.keys(functions).forEach(functionName => {
-    setStoredKeyValue(context, functionName, functions);
+    setStoredKeyValue(context, functionName, functions[functionName]);
   });
 };
 
+export const askEnvironmentVariableToPush = async (context: $TSContext, currentEnvName: string, projectPath: string) => {
```

Comment:
I'd rename this to something like `ensureEnvironmentVariableValues` because if everything is already defined, it's not going to ask anything

Reply:
Agreed on the better name you suggested! I'll update.